File: app-core/src/main/java/ru/micron/dto/omdb/ApiSearchResponse.java
package ru.micron.dto.omdb;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ApiSearchResponse {

  @JsonProperty("Search")
  private List<Search> result;

  @JsonProperty("totalResults")
  private String totalAmount;

  @JsonProperty("Response")
  private Boolean response;

  @Data
  @AllArgsConstructor
  @NoArgsConstructor
  @JsonIgnoreProperties(ignoreUnknown = true)
  public static class Search {
    @JsonProperty("Title")
    private String title;

    @JsonProperty("Year")
    private String year;

    @JsonProperty("imdbID")
    private String imdbId;

    @JsonProperty("Type")
    private String type;

    @JsonProperty("Poster")
    private String posterUrl;
  }
}

File: app-core/src/main/kotlin/ru/micron/dto/omdb/ApiSearchResponse.kt
package ru.micron.dto.omdb

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class ApiSearchResponse(
    @JsonProperty("Search")
    var result: List<Search>? = null,
    @JsonProperty("totalResults")
    var totalAmount: String? = null,
    @JsonProperty("Response")
    var response: Boolean? = null
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Search(
        @JsonProperty("Title")
        var title: String? = null,
        @JsonProperty("Year")
        var year: String? = null,
        @JsonProperty("imdbID")
        var imdbId: String? = null,
        @JsonProperty("Type")
        var type: String? = null,
        @JsonProperty("Poster")
        var posterUrl: String? = null
    )
}
